src/hooks/useQuery.ts

```diff
@@ -1,5 +1,12 @@
+import React from 'react';
 import { useLocation } from 'react-router-dom';
 
 export const useQuery = () => {
-  return new URLSearchParams(useLocation().search);
+  const search = useLocation().search;
+
+  const query = React.useMemo(() => {
+    return new URLSearchParams(search);
+  }, [search]);
+
+  return query;
 };
```

Comment:
Maybe you could rename this hook e.g. useSearchParams. At first time when I read this PR through I wasn't sure what it does

Reply:
renamed